pkg/maintainer/spv/spv.go

```diff
@@ -59,8 +63,271 @@ func (sm *spvMaintainer) startControlLoop(ctx context.Context) {
+	// There will often be multiple events emitted for a single wallet. Prepare
+	// a list of unique wallet public key hashes.
+	walletPublicKeyHashes := uniqueWalletPublicKeyHashes(
+		depositSweepTransactionProposals,
+	)
+
+	unprovenDepositSweepTransactions := []*bitcoin.Transaction{}
+
+	for _, walletPublicKeyHash := range walletPublicKeyHashes {
+		// TODO: Should we check the wallet's state before attempting to submit
+		//       the deposit sweep proof?
+		// TODO: Think what the limit of transactions retrieved should be.
+		walletTransactions, err := sm.btcChain.GetTransactionsForPublicKeyHash(
+			walletPublicKeyHash,
+			5,
+		)
+		if err != nil {
+			return nil, fmt.Errorf(
+				"failed to get transactions for wallet: [%v]",
+				err,
+			)
+		}
+
+		for _, transaction := range walletTransactions {
+			isUnprovenDepositSweepTransaction, err :=
+				sm.isUnprovenDepositSweepTransaction(
+					transaction,
+					walletPublicKeyHash,
+				)
+			if err != nil {
+				return nil, fmt.Errorf(
+					"failed to check if transaction is an unproven deposit sweep "+
+						"transaction: [%v]",
+					err,
+				)
+			}
+
+			if isUnprovenDepositSweepTransaction {
+				unprovenDepositSweepTransactions = append(
+					unprovenDepositSweepTransactions,
+					transaction,
+				)
+			}
+		}
+	}
+
+	return unprovenDepositSweepTransactions, nil
+}
+
+func (sm *spvMaintainer) isUnprovenDepositSweepTransaction(
+	transaction *bitcoin.Transaction,
+	walletPublicKeyHash [20]byte,
+) (bool, error) {
+	// If the transaction does not have exactly one output, it cannot be a
+	// deposit sweep transaction.
+	if len(transaction.Outputs) != 1 {
+		return false, nil
+	}
+
+	hasDepositInputs := false
+
+	// Look at the transaction's inputs. All the inputs must be deposit inputs,
+	// except for one input which can be the main UTXO.
+	for _, input := range transaction.Inputs {
+		fundingTransactionHash := input.Outpoint.TransactionHash
+		fundingOutpointIndex := input.Outpoint.OutputIndex
+
+		// Check if the input is a deposit input.
+		deposit, err := sm.chain.Deposits(
+			fundingTransactionHash,
+			fundingOutpointIndex,
+		)
+		if err != nil {
+			return false, fmt.Errorf("failed to get a deposit: [%v]", err)
+		}
+
+		if deposit.RevealedAt.Equal(time.Unix(0, 0)) {
+			// The input is not a deposit input. The transaction can still be
+			// a deposit sweep transaction, since the input may be the main UTXO.
+
+			// Check if the input represents the current main UTXO of the wallet.
+			// Notice that we don't have to verify if there is only one main
+			// UTXO among the transaction's inputs since only one input may have
+			// such a structure that the calculated hash will match the wallet's
+			// main UTXO hash stored on-chain.
+			isMainUtxo, err := sm.isInputCurrentWalletsMainUTXO(
+				fundingTransactionHash,
+				fundingOutpointIndex,
+				walletPublicKeyHash,
+			)
+			if err != nil {
+				return false, fmt.Errorf("failed to check is input is the main UTXO")
```

Comment:
```suggestion
				return false, fmt.Errorf("failed to check if input is the main UTXO")
```

Reply:
Done in 28e3faa5fa4eda8ca3f2f451f60ac5889aa2cd40.